File: ReservationFrontend/src/models/Reservation.ts
import { ReservationStatus } from "./enums/ReservationStatus";

export interface Reservation {
  id: number;
  customerUsername?: string;
  vehicleId: number;
  licensePlate: string;
  vin: string;
  brand: string;
  model: string;
  year: string;
  carModelId: number;
  creationDate: Date;
  plannedPickUpDate: Date;
  actualPickUpDate?: Date;
  plannedDropOffDate: Date;
  actualDropOffDate?: Date;
  status: ReservationStatus;
  totalAmount: number;
  // Boolean attributes used to compute eligibility score
  wasDeliveryLate?: boolean;
  wasChargedFee?: boolean;
  wasVehicleDamaged?: boolean;
  wasInvolvedInAccident?: boolean;
}

File: AnalyticsService/src/main/kotlin/com/rentalcarsystem/analyticsservice/models/Reservation.kt
package com.rentalcarsystem.analyticsservice.models

import com.rentalcarsystem.analyticsservice.enums.ReservationStatus
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "reservations")
class Reservation(
    @Column(nullable = false)
    var customerUsername: String,

    @Column(nullable = false)
    var creationDate: LocalDateTime,

    @Column(nullable = false)
    var plannedPickUpDate: LocalDateTime,

    var actualPickUpDate: LocalDateTime? = null,

    @Column(nullable = false)
    var plannedDropOffDate: LocalDateTime,

    var actualDropOffDate: LocalDateTime? = null,

    var bufferedDropOffDate: LocalDateTime,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var status: ReservationStatus,

    @Column(nullable = false)
    var totalAmount: Double,

    // Boolean attributes used to compute eligibility score
    var wasDeliveryLate: Boolean? = null,
    var wasChargedFee: Boolean? = null,
    var wasInvolvedInAccident: Boolean? = null,

    // Integer attributes ranging from 0 to 5 (0 = best case, 5 = worst case)
    var damageLevel: Int? = null,
    var dirtinessLevel: Int? = null,

    // Staff members who handled the reservation
    var pickUpStaffUsername : String?=null,
    var dropOffStaffUsername : String?=null,
    var updatedVehicleStaffUsername : String?=null,

    ) : BaseEntity<Long>()